utils.go

```diff
@@ -62,6 +62,58 @@ func DecodeLabel(image imgutil.Image, label string, v interface{}) error {
 	return nil
 }
 
+// RemoveStagePrefixes creates a slice with the build: or run: style prefixes removed.
+func RemoveStagePrefixes(mixins []string) []string {
+	var result []string
+	for _, m := range mixins {
+		s := strings.SplitN(m, ":", 2)
+		if len(s) == 1 {
+			result = append(result, s[0])
+		} else {
+			result = append(result, s[1])
+		}
+	}
+	return result
+}
+
+// FromSlice converts the given slice to a set in the form of unique keys in a map.
+// The value associated with each key should not be relied upon. A value is present
+// in the set if its key is present in the map, regardless of the key's value.
+// from: https://github.com/buildpacks/pack/blob/main/internal/stringset/stringset.go
+func FromSlice(strings []string) map[string]interface{} {
+	set := map[string]interface{}{}
+	for _, s := range strings {
+		set[s] = nil
+	}
```

Comment:
```suggestion
func FromSlice(strings []string) map[string]struct{} {
	set := map[string]struct{}{}
	for _, s := range strings {
		set[s] = struct{}
	}
```
It is typical to use a struct for this since it takes zero bytes but some people also use `map[T]bool` https://www.davidkaya.com/sets-in-golang/.